File: springboot-backend/src/main/java/backend/dto/response/ProductResponse.java
package backend.dto.response;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductResponse {
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private Integer discount;
    private Float rating;
    private Integer sold;
    private Integer stock;
    private String image;
    private String shopName;
    private String eta;
    private Long categoryId;
    private String categoryName;
    private Integer soldQuantity;
    private Double averageStars;

    private CategoryResponse category;

    public void setPrice(Double priceValue) {
        this.price = priceValue != null ? BigDecimal.valueOf(priceValue) : BigDecimal.ZERO;
    }
    
    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public void setRating(Double ratingValue) {
        this.rating = ratingValue != null ? ratingValue.floatValue() : 0.0f;
    }

    public void setRating(Float rating) {
        this.rating = rating;
    }
}

File: Fontend/app/src/main/java/com/example/data/dto/response/ProductResponse.kt
package com.example.data.dto.response

import com.google.gson.annotations.SerializedName

data class ProductResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String?,
    @SerializedName("description") val description: String?,
    @SerializedName("price") val price: Double?,
    @SerializedName("image") val image: String?,
    @SerializedName("rating") val rating: Double?,
    @SerializedName("sold") val sold: Int?,
    @SerializedName("category") val category: CategoryResponse?
)